### DISCUSSION TITLE
[Scroll Gesture] Track target's position relative to non-window container

### DISCUSSION THREAD

**Discussion by ghost** [motiondivision/motionone#148] [Support]
Hi, Is it possible to track the position of the target relative to the specified container instead of the viewport? For example, I want to create a horizontal slider with multiple slides. And I want to track the entry and exit of each slide relative to the slider instead of viewport.

```typescript
const slider = document.querySelector<HTMLElement>('.js-slider')!;
const slides = slider.querySelectorAll<HTMLElement>('.js-slide')!;

slides.forEach((slide) => {
  scroll(
    // Instead of this callback will be animate fn.
    (info) => {
      console.log('info', ':', info.x.progress);
    },
    {
      target: slide,
      container: slider,
      offset: [...ScrollOffset.Enter, ...ScrollOffset.Exit],
      axis: 'x',
    }
  );
});
```

Demo: https://stackblitz.com/edit/vitejs-vite-kfptsx?file=src%2Findex.ts

**Comment by mattgperry** [comment]
Yeah this should be possible as you have it. I’d be interested in seeing a sandbox if it’s not working for you.

**Reply by ghost** [reply]
The offsets seem to be relative to the window.
